toolchain/gcc-14.2.0-13-arm-multilib.patch

```diff
@@ -0,0 +1,208 @@
+diff --git a/gcc/config.gcc b/gcc/config.gcc
+--- a/gcc/config.gcc
++++ b/gcc/config.gcc
+@@ -1475,7 +1475,7 @@ arm*-*-phoenix*)
+ 	tm_file="elfos.h arm/unknown-elf.h arm/elf.h arm/bpabi.h"
+ 	tm_file="${tm_file} newlib-stdint.h phoenix.h"
+ 	tm_file="${tm_file} arm/aout.h arm/arm.h"
+-	tmake_file="${tmake_file} arm/t-arm arm/t-bpabi arm/t-aprofile arm/t-rmprofile arm/t-multilib"
++	tmake_file="${tmake_file} arm/t-arm arm/t-bpabi arm/t-multilib arm/t-phoenix"
+ 	with_float="soft" # "auto" fpu
+ 	TM_MULTILIB_CONFIG="aprofile,rmprofile"
+ 	target_cpu_cname="arm7tdmi"
+diff --git a/gcc/config/arm/t-multilib b/gcc/config/arm/t-multilib
+index be63edd85c4..14c6c9875f1 100644
+--- a/gcc/config/arm/t-multilib
++++ b/gcc/config/arm/t-multilib
+@@ -225,4 +225,3 @@ MULTILIB_REUSE		+= $(foreach ABI, hard softfp, \
+ MULTILIB_REUSE		+= $(foreach MODE, arm thumb, \
+ 			     $(foreach ARCH, armv7 armv7-r armv8-r, \
+ 			       mthumb/march.$(ARCH)/mfloat-abi.soft=m$(MODE)/march.$(ARCH)/mfloat-abi.softfp))
+-
+diff --git a/gcc/config/arm/t-phoenix b/gcc/config/arm/t-phoenix
+--- a/gcc/config/arm/t-phoenix
++++ b/gcc/config/arm/t-phoenix
+@@ -16,16 +16,45 @@
+ # along with GCC; see the file COPYING3.  If not see
+ # <http://www.gnu.org/licenses/>.
+ 
+-MULTILIB_OPTIONS	= marm/mthumb
+-MULTILIB_DIRNAMES	= arm thumb
+-MULTILIB_EXCEPTIONS	=
+-MULTILIB_MATCHES	=
+-MULTILIB_REUSE		=
+-MULTILIB_REQUIRED	=
+-
+-MULTILIB_OPTIONS	+= mfloat-abi=hard
+-MULTILIB_DIRNAMES	+= fpu
+-MULTILIB_MATCHES	+= mfloat-abi?hard=mhard-float
+-
+-MULTILIB_OPTIONS	+= mno-thumb-interwork/mthumb-interwork
+-MULTILIB_DIRNAMES	+= normal interwork
++# Add phoenix options to reuses and requires.
++
++s-mlib: $(srcdir)/config/arm/t-phoenix
++
++# Find all RM archs.
++RM_ARCHS			:= $(subst /, ,march=armv7-r+fp.sp/$(MULTI_ARCH_OPTS_RM))
```

Comment:
why this one arch needs to be listed here?

Reply:
For some reason(no idea why) GCC adds it separately in t-mulitlib instead of t-rmprofile
https://github.com/gcc-mirror/gcc/blob/0d998b38a6fd280973fa6e91d3e34e544eef8fb2/gcc/config/arm/t-multilib#L95

Reply:
ack. I've read commit related to the `git blame` - seems it makes some sense due to merging some of the subvariants, but I did not dig into the actual mechanism too match.

Thanks for the info.